subprojects/core/src/main/java/org/gradle/api/internal/tasks/userinput/UserInputHandler.java

```diff
@@ -31,86 +34,17 @@ public interface UserInputHandler {
      */
     @Nullable
     @UsedByScanPlugin
-    Boolean askYesNoQuestion(String question);
-
-    /**
-     * Prompts the user with a question that has a boolean result and returns the answer.
-     *
-     * <p>The client UI is free to choose a convenient representation for boolean values, for example
-     * allowing the user to type 'y' or 'n' or presenting a checkbox. The user is not required to answer 'true' or 'false'
-     *
-     * @param question The text of the question.
-     * @param defaultValue The option to present to the user as the default choice, and the value to use when not connected to a console
-     * @return the answer or the given default if not connected to a console.
-     */
-    boolean askBooleanQuestion(String question, boolean defaultValue);
-
-    /**
-     * Prompts the user to select an option from the given list and returns the answer.
-     * Uses the {@link Object#toString()} representation of the options to format the prompt.
-     * Does not prompt the user when there is only one option in the given list.
-     *
-     * @param question The text of the question.
-     * @param defaultOption The option to present to the user as the default selection, and also the value to use when not connected to a console.
-     * @return the answer or the default if not connected to a console.
-     */
-    <T> T selectOption(String question, Collection<T> options, T defaultOption);
-
-    /**
-     * Prompts the user to select an option from the given list and returns the answer.
-     * Uses the {@link Object#toString()} representation of the options to format the prompt, the returned builder can be used to change this.
-     * Does not prompt the user when there is only one option in the given list.
-     *
-     * @param question The text of the question.
-     * @return A builder that can be used to configure the choice and ask the user to select an option.
-     */
-    <T> ChoiceBuilder<T> choice(String question, Collection<T> options);
-
-    /**
-     * Prompts the user with a question that has an integer result and returns the answer.
-     *
-     * @param question The text of the question.
-     * @param defaultValue The option to present to the user as the default choice, and the value to use when not connected to a console
-     * @return The answer or the given default if not connected to a console.
-     */
-    int askIntQuestion(String question, int minValue, int defaultValue);
+    default Boolean askYesNoQuestion(String question) {
+        return askUser(interaction -> interaction.askYesNoQuestion(question)).get();
```

Comment:
❌ This breaks scans when the user doesn't supply the yes or no as response. I should be `getOrNull()`.

Reply:
Fix is in https://github.com/gradle/gradle/pull/28015